### PR TITLE
Add cookbook recipe: Kirby meets TailwindCSS

### PR DESCRIPTION
**Pull request by chrfickinger**

### CODE REVIEW

**Comment by texnixe** [comment]
Hi Christian, thanks for your PR.

Apart from a couple of typos and a package.json that shouldn't have comments in it, my main concern is that the setup deviates from the usual ones, e.g. described [here](https://www.brianliddell.com/blog/how-to-set-up-local-development-for-kirby-and-tailwind-using-the-command-line-on-macos)  or [here](https://sebastian-widmann.de/tailwind-css-mit-dem-kirby-cms-nutzen/) or in the official docs. In particular, I think the source styles should never be in the same location as the dist styles.

Is there a reason why you chose this setup?

**Comment by chrfickinger** [comment]
Thank you for the feedback Sonja and sorry for the typos. Of course that shouldn't happen, obviously I was more busy understanding GIT than proofreading. I also didn't know that you should avoid comments in packages.json. I thought it helps newbies to understand.

Actually, there is no particular reason why the generated output file is in the same directory as the source. Of course we can change it to _/dist_, _/assets_ or _/generated_. I just wanted to show that in the end it's all about these files, which are needed for a basic installation. I have updated the recipe, please review again.

File: content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt [hunk 1]
```diff
@@ -0,0 +1,119 @@
+Title: Kirby meets Tailwind CSS
+
+----
+
+Text:
+
+This is a walktrouh on how to install and use Tailwind CSS with Kirby.
```
Comment (texnixe) [inline content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt]: walkthrough

File: content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Before we start, we need to install Node.js for the build process on your client machine (or on the machine where you want to use Tailwind CSS). The latest version, which also includes npm (Node package management) can be downloaded on the (link: https://nodejs.org/en/download/ text: official Node.js website). You can test the successful installation by opening a terminal window and typing the following two commands.
+
+```bash
+node --version
+```
+```bash
+npm --version
+```
+
+We use a plain kirby installation to get started and just add a new folder `build` and a new folder `assets` to the root directory.
```
Comment (texnixe) [inline content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt]: Kirby

File: content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+Use `build` to generate a final minified css file:
+```bash
+npm run build
+```
+
+Both scripts check all php files inside the `site` directory for Tailwind CSS classes and writes them to the css file `site.css`. You just need to add the css source to your HTML markup like follows:
+```php
+<?= css('assets/style.css') ?>
```
Comment (texnixe) [inline content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt]: styles.css

File: content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+
+
+
+
+
+----
+
+Description: A walktrouh on how to install and use Tailwind CSS with Kirby.
```
Comment (texnixe) [inline content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt]: walkthrough

File: content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+<info>This is just a basic setup. More details on how to customize Tailwind CSS can be found on the (link: https://tailwindcss.com/ text:Tailwind CSS website)</info>
+
+After creating the build configuration, we create a basic css file and use the `@tailwind` directive to inject Tailwind’s `base`, `components`, and `utilities` styles.
+```css "/build/tailwind.css"
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+```
+
+Our `build`directory should look like follows and we are ready to install Tailwind CSS.
```
Comment (texnixe) [inline content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt]: whitespace missing after `build`

File: content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+README.md
+site/
+```
+
+## Configuration
+Now let's create the file `package.json` which controls our build process inside the newly created directory. This file contains two build scripts, one for development (*watch*) and one for live mode (*build*).
+
+```js "/build/package.json"
+{
+    "name": "My project",
```
Comment (texnixe) [inline content/docs/2_cookbook/9_setup/0_kirby-meets-tailwindcss/cookbook-recipe.txt]: Space in name not allowed

**Review by texnixe — COMMENTED** [review]
In general, I'm still somehow missing  how this would work with config options. Isn't being able to configure the base settings something I would always want to do when using Tailwind? Since I have never used Tailwind apart from one short try, I can't really tell but would expect that.

@getkirby/kirby-staff, what do you think?

**Comment by chrfickinger** [comment]
Just to note: This is the absolute basic setup and I kept it simple on purpose. More config settings (tailwind.config.js, etc) are of course possible, but actually not necessary to get started. You should dive into the official tailwind documentation anyway, if you want to customize and configure the css framework.

**Review by texnixe — APPROVED** [review]

**Comment by bastianallgeier** [comment]
I love that it is so compact although I agree that it would be cool just to mention where to put the config and give a quick example how a small adjustment would look like. But I leave it up to you both if you want to add it before your release it.

**Review by bastianallgeier — APPROVED** [review]

**Comment by chrfickinger** [comment]
> I love that it is so compact although I agree that it would be cool just to mention where to put the config and give a quick example how a small adjustment would look like. But I leave it up to you both if you want to add it before your release it.

Good point. I'll add a block about optional config options.

**Pull request merged**